src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTokenExtensions.cs

```diff
@@ -170,6 +174,19 @@ public static bool IsBeginningOfStatementContext(this SyntaxToken token)
 
                 case SyntaxKind.ElseKeyword:
                     return true;
+
+                case SyntaxKind.CloseBracketToken:
+                    if (token.Parent.IsKind(SyntaxKind.AttributeList))
+                    {
+                        // attributes can belong to a statement
+                        var container = token.Parent.Parent;
+                        if (container is StatementSyntax)
```

Comment:
Can this be a non local function statement in valid code?

Reply:
No, it has to be a local function statement to be valid.